React/Views/RCTModalHostView.m

```diff
@@ -54,6 +61,27 @@ - (instancetype)initWithBridge:(RCTBridge *)bridge
   return self;
 }
 
+#if TARGET_OS_TV
+- (void)menuButtonPressed:(__unused UIGestureRecognizer *)r
+{
+    if(_onRequestClose) {
+        _onRequestClose(nil);
+    }
+}
+
+- (void)setOnRequestClose:(RCTDirectEventBlock)onRequestClose
+{
+  _onRequestClose = onRequestClose;
+  if([self.subviews count] > 0) {
+    if(_onRequestClose && _tapGestureRecognizer) {
+      [self.subviews[0] addGestureRecognizer:_tapGestureRecognizer];
```

Comment:
Using `subviews[]` here is technically correct, but it is extremly fragile abstraction. In better world we would have dedicated property, something like `contentView`, which smart getter and setter would manage gesture recognizers and add/remove it from view hierarchy.